### PR TITLE
Get rid of PowHSMSignerMessage legacy constructor and simplify tests

### PR DESCRIPTION
**Pull request by nathanieliov**
- Get rid of PowHSMSignerMessage legacy constructor
- Update tests

## Description

- Get rid of the PowHSMSignerMessage legacy constructor
- Simplify and reorganize tests
- Moved duplicated code to helper private methods

## Motivation and Context

Remove the PowHSMSignerMessage legacy constructor, as it is no longer needed. Also, update tests to make them simpler.

## How Has This Been Tested?

Unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Tests for the changes have been added (for bug fixes / features)

### CODE REVIEW

File: src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageBuilderTest.java [hunk 1]
```diff
@@ -68,12 +70,14 @@ class PowHSMSignerMessageBuilderTest {
         bridgeMainnetConstants.getBtcParams(), 9);
     private static final Federation oldFederation = TestUtils.createFederation(
         bridgeMainnetConstants.getBtcParams(), 9);
-
+    private final BtcECKey signerBtcPk = BtcECKey.fromPrivate(
+        org.bouncycastle.util.encoders.Hex.decode("fa01"));
```
Comment (apancorb) [inline src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageBuilderTest.java:74]: Why not use the `decode` function from `org.spongycastle.util.encoders.Hex` here?

File: src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageTest.java [hunk 2]
```diff
@@ -67,51 +71,30 @@ class PowHSMSignerMessageTest {
...
-        ObjectNode expectedMessageVersion5 = new ObjectMapper().createObjectNode();
-        expectedMessageVersion5.put(TX.getFieldName(), expectedEncodedPegoutBtcTx);
-        expectedMessageVersion5.put(INPUT.getFieldName(), FIRST_OUTPUT_INDEX);
-        expectedMessageVersion5.put(SIGHASH_COMPUTATION_MODE.getFieldName(), SIGHASH_LEGACY_MODE);
-        arguments.add(Arguments.of(hsmVersion5, expectedMessageVersion5));
+    @BeforeEach
+    void setUp() {
 
-        return arguments;
     }
```
Comment (apancorb) [inline src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageTest.java:89]: Do we need this `setUp` method?

**Review by apancorb — APPROVED** [review]
LGTM, left two minor comments

File: src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageTest.java (hunk 2 above)
Comment (nathanieliov) [reply src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageTest.java]: Good catch.

File: src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageBuilderTest.java (hunk 1 above)
Comment (nathanieliov) [reply src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageBuilderTest.java]: Good catch!

File: src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageTest.java (hunk 2 above)
Comment (nathanieliov) [reply src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageTest.java]: Solved

File: src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageBuilderTest.java [hunk 3]
```diff
@@ -526,9 +430,15 @@ private BtcTransaction createPegout(List<Coin> outpointValues, Address destinati
         return pegoutBtcTx;
     }
 
-    private void addWitness(BtcTransaction pegoutBtcTx, int inputIndex) {
-        TransactionWitness txWitness = new TransactionWitness(1);
-        txWitness.setPush(0, new byte[]{0x1});
-        pegoutBtcTx.setWitness(inputIndex, txWitness);
+    private void addCommonPegoutLogs(List<LogInfo> logs, BtcTransaction pegoutBtcTx) {
```
Comment (marcos-iov) [inline src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageBuilderTest.java:433]: Shouldn't all transactions have batch_pegout_created event?

Comment (nathanieliov) [reply src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageBuilderTest.java:433]: It shouldn't when the pegout is a migration.

**Pull request merged**